Dockerfile

```diff
@@ -1,52 +1,65 @@
-# base node image
-FROM node:16-bullseye-slim as base
-
-# set for base and all layer that inherit from it
-ENV NODE_ENV production
-
-# Install openssl for Prisma
-RUN apt-get update && apt-get install -y openssl
+FROM node:18-bullseye-slim@sha256:e2fbe082615911b184e192b05c55e7e38460a2c24c88d92e8c122ea0175fbe56 as base
```

Comment:
Is there a reason we pin to a commit?

Reply:
If I remember from CCP David likes to do this because he knows exactly which version of the image will work for the app and doesn't want it to be left to the latest for example because that is subject to potentially breaking changes.

Reply:
Best practices for deploying Docker images includes pinning your dependencies so you have reproducible environments.  If you deploy to staging with version 0.0.1 and then by the time you deploy to production it's 0.0.2, you are not deploying the same software.

It *should* be true that people only do breaking changes with a major update; but "should" isn't what happens.  So you end-up having your production boxes breaking because one version of something is slightly off.

The answer is to pin your images, dependencies, modules, etc.